packages/api/src/utils/db-utils.ts

```diff
@@ -0,0 +1,25 @@
+import { AnySelectQueryBuilder } from "kysely"
+import { z } from "zod"
+
+export const CursorSchema = z.object({
+  id: z.string(),
+  createdAt: z.date(),
+})
+
+export const PaginateInputSchema = z
+  .object({
+    take: z.number(),
+    cursor: CursorSchema.optional(),
+  })
+  .optional()
+  .default({ take: 20, cursor: undefined })
+
+export type Cursor = z.infer<typeof CursorSchema>
+
+export const paginateQuery = (qb: AnySelectQueryBuilder, cursor: Cursor) => {
+  return qb
+    .where((qb) => qb.where("createdAt", "=", cursor.createdAt).where("id", "<", cursor.id))
```

Comment:
How does this work? Doesn't this become uuid comparison?

Reply:
I just did this https://dba.stackexchange.com/questions/267794/how-to-do-pagination-with-uuid-v4-and-created-time-on-concurrent-inserted-data

Reply:
![image](https://user-images.githubusercontent.com/41551253/222925133-7dab155d-1466-49c7-8a5f-e50437ac2e75.png)